sdk/azcore/arm/runtime/pipeline.go

```diff
@@ -12,30 +12,28 @@ import (
 	armpolicy "github.com/Azure/azure-sdk-for-go/sdk/azcore/arm/policy"
 	"github.com/Azure/azure-sdk-for-go/sdk/azcore/internal/pipeline"
 	"github.com/Azure/azure-sdk-for-go/sdk/azcore/internal/shared"
-	azpolicy "github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
 	azruntime "github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
 )
 
 // NewPipeline creates a pipeline from connection options.
 // The telemetry policy, when enabled, will use the specified module and version info.
-func NewPipeline(module, version string, cred azcore.TokenCredential, options *arm.ClientOptions) pipeline.Pipeline {
+func NewPipeline(module, version string, cred azcore.TokenCredential, plOpts azruntime.PipelineOptions, options *arm.ClientOptions) pipeline.Pipeline {
 	if options == nil {
 		options = &arm.ClientOptions{}
 	}
 	ep := options.Host
 	if len(ep) == 0 {
 		ep = arm.AzurePublicCloud
 	}
-	perCallPolicies := []azpolicy.Policy{}
+	authPolicy := NewBearerTokenPolicy(cred, &armpolicy.BearerTokenOptions{
+		Scopes:           []string{shared.EndpointToScope(string(ep))},
+		AuxiliaryTenants: options.AuxiliaryTenants,
+	})
+	o := plOpts
+	o.PerRetry = append(o.PerRetry, authPolicy)
```

Comment:
Depending on the capacity of `PerRetry` this can modify the underlying array.  I put together an example the demonstrates this [here](https://play.golang.org/p/KMu97L8xuVw).

To avoid potentially mutating the backing array of a slice, we should just create new arrays that merge the inputs together.